superduperdb/base/superduper.py

```diff
@@ -24,26 +24,16 @@ def superduper(item: t.Optional[t.Any] = None, **kwargs) -> t.Any:
 
 
 def _auto_identify_connection_string(item: str, **kwargs) -> t.Any:
-    from superduperdb.base.build import build_datalayer
-
-    if item.startswith('mongomock://'):
-        kwargs['data_backend'] = item
+    if re_match(r'^[a-zA-Z0-9]+://', item) is None:
+        raise ValueError(f'{item} is not a valid connection string')
 
-    elif item.startswith('mongodb://'):
-        kwargs['data_backend'] = item
+    if item.endswith('.csv') and CFG.cluster.cdc.uri is not None:
+        raise TypeError('Pandas is not supported in cluster mode!')
 
-    elif item.startswith('mongodb+srv://') and 'mongodb.net' in item:
```

Comment:
Hey @ChandanChainani , we need to check this pattern, because `re_match(r'^[a-zA-Z0-9]+://', 'mongodb+srv://') is None == Ture`, then will raise an error.

Reply:
@jieguangzhou for this type of conditions we can improve by creating single condition which validates all the supported url types.

For example:
```py
supported_url_types = ["mongodb+srv://", "mongodb://", "mongomock://"]
if not any(item.startswith(url_type) for url_type in supported_url_types):
    raise Exception("url format not supported")
```

Reply:
Yes, could you update the pr with these changes?